terminator_test.go

```diff
@@ -277,6 +279,39 @@ func TestTerminator_Wait(t *testing.T) {
 	}
 }
 
+func TestTerminator_restores_signals(t *testing.T) {
+	if os.Getenv("HANG_ON_STOP") == "1" {
+		terminator, ctx := NewWithSignals(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+		require.NotNil(t, ctx)
+
+		terminator.WithOrder(0).Register(5*time.Second, func(ctx context.Context) {
+			select {}
+		})
+
+		_ = terminator.Wait(ctx, time.Minute)
+		return
+	}
+
+	cmd := exec.Command(os.Args[0], "-test.run=TestTerminator_restores_signal")
```

Comment:
it's not a good approach to introduce a recursion in tests, pls see the comment above

Reply:
Subprocess-based testing is a widely used technique
You can find a lot of articles mentions it. I.e. https://about.sourcegraph.com/blog/go/advanced-testing-in-go

Reply:
> @BMalaichik thank you for review! I implemented "preStop" because k8s' one relies on "sleep" availability inside container. But sometimes container doesn't contain it. I. e. container built from scratch with app only.

Hm, `sleep` is provided as a part of `coreutils` distribution package, having a size <3 MBs. I believe if the docker container is packed as minimalistic as possible for some reason `graterm` not likely to be used there as a dependency either :)

Reply:
"Distroless" images from Google doesn't contain coreutils.
```bash
$ docker run --rm -it gcr.io/distroless/base sleep 1
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "sleep": executable file not found in $PATH: unknown.
```

Reply:
Thinking generally about this approach I believe it makes sense when you do integration testing of your CLI application, so it's easy an interesting approach to bind the test case to the CLI program execution, as in the given article they do in Hashicorp

For this particular case testing a package providing a utility structure it's an overkill IMO